substratevm/src/com.oracle.svm.jfr/src/com/oracle/svm/jfr/JfrThreadLocal.java

```diff
@@ -205,6 +205,23 @@ public JfrBuffer getNativeBuffer() {
         return result;
     }
 
+    @Uninterruptible(reason = "Accesses a JFR buffer.")
+    public static JfrBuffer getJavaBuffer(IsolateThread thread) {
```

Comment:
Please assert that this method is only executed in a safepoint. `callerMustBe = true` can be added to the `@Uninterruptible` annotation.